heartbeat/asterisk

```diff
@@ -345,6 +345,15 @@ asterisk_start() {
       for i in $astcanary_pid; do ocf_run kill -s KILL $astcanary_pid; done
     fi
 
+    for dir in $ASTRUNDIR $ASTLOGDIR $ASTLOGDIR/cdr-csv $ASTLOGDIR/cdr-custom; do
+	# Check if directories are writable and fix them if they aren't
+        if  ! su -s /bin/sh - $OCF_RESKEY_user -c "test -w $dir" ; then
+            ocf_log warn "Directory $dir is not writable by $OCF_RESKEY_user, attempting to fix."
+			mkdir -p $dir
+			chown $OCF_RESKEY_user: $dir
```

Comment:
You should move your chown line here, and the creation is being done by `install -d` just above it:
https://github.com/ClusterLabs/resource-agents/pull/1583#diff-1fc5dbf5fe478c7910a2675238f986c1011c8250e498f648963c7fb3cb0cd342R365

You should also set group with chown, as there's a parameter for that in the agent too.

Reply:
Not sure where you meant I should move the chown line.  If I just move it inside the "-d" conditional check it will not be executed as long if the directory exists, but is not writable with the designated user, so it will fail.
As for the group - OK, I will correct it.
Can you please clarify?
Thanks.

Reply:
Inside the `if ! su -s /bin/sh - $OCF_RESKEY_user -c "test -w $dir"; then` on line 365.

Reply:
Thanks, commited changes.